website/events/templates/events/event.html

```diff
@@ -132,15 +132,19 @@ <h1 class="text-center section-title">{{ event.title }}</h1>
                                         <td>
                                             {% if registration.is_paid %}
                                                 {% if registration.payment.type == payment_method_tpay %}
-                                                    {% trans "You pay with Thalia Pay" %}
+                                                    {% trans "You are paying with Thalia Pay" %}
                                                 {% else %}
                                                     {% trans "You have paid" %}
                                                 {% endif %}
                                             {% else %}
                                                 {% trans "You have not paid yet" %}
                                                 {% if registration.member.tpay_enabled %}
-                                                    <form action="{% url 'events:pay' event.id %}" method="post">
+                                                    <form action="{% url 'payments:payment-process' %}" method="post">
                                                         {% csrf_token %}
+                                                        <input type="hidden" name="model_name" value="eventregistration" />
+                                                        <input type="hidden" name="app_label" value="events" />
+                                                        <input type="hidden" name="payable" value="{{ registration.pk }}" />
+                                                        <input type="hidden" name="next" value="{{ request.path }}"/>
                                                         <input type="submit" class="btn btn-primary"
                                                                value="{% trans "Pay with Thalia Pay" %}"/>
                                                     </form>
```

Comment:
Maybe, if you're doing this, make this a template tag in the payments app too

Reply:
that is something I considered indeed, but I was lazy